### PR TITLE
add assisted inject

### PR DESCRIPTION
**Pull request by yehezkiell**
## Guidelines
Hi skydoves, this is great work. I'm adding asssited inject in `DetailViewModel` so the view no need to assign pokemon name by function instead of using constructor

### Types of changes
What types of changes does your code introduce?
feature improvement

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Preparing a pull request for review
Ensure your change is properly formatted by running:

```gradle
$ ./gradlew spotlessApply
```

Please correct any failures before requesting a review.

### CODE REVIEW

File: app/src/main/java/com/skydoves/pokedex/ui/details/DetailViewModel.kt [hunk 1]
```diff
@@ -16,45 +16,55 @@
...
   val toastLiveData: MutableLiveData<String> = MutableLiveData()
+  val pokemonInfoLiveData: LiveData<PokemonInfo?> = launchOnViewModelScope(
+    block = {
+      isLoading.set(true)
+      detailRepository.fetchPokemonInfo(
+        name = pokemonName,
+        onSuccess = { isLoading.set(false) },
+        onError = { toastLiveData.postValue(it) }
+      ).asLiveData()
+    }
```
Comment (skydoves) [inline app/src/main/java/com/skydoves/pokedex/ui/details/DetailViewModel.kt:47]: I think it is better that initialize in init block for consistency with the MainViewModel. 

```kotlin
  val pokemonInfoLiveData: LiveData<PokemonInfo?>

  init {
    Timber.d("init DetailViewModel")

    pokemonInfoLiveData = launchOnViewModelScope {
      isLoading.set(true)
      detailRepository.fetchPokemonInfo(
        name = pokemonName,
        onSuccess = { isLoading.set(false) },
        onError = { toastLiveData.postValue(it) }
      ).asLiveData()
    }
  }
```

**Review by skydoves — COMMENTED** [review]
Hi, thank you for your contribution!
I just reviewed and left a comment.
This PR is a good example of `AssistedInject` and `Hilt` working together. 👍

File: app/src/main/java/com/skydoves/pokedex/ui/details/DetailViewModel.kt (hunk 1 above)
Comment (yehezkiell) [reply app/src/main/java/com/skydoves/pokedex/ui/details/DetailViewModel.kt]: Hi @skydoves thanks for the review, I've changed the code plus fix the `DetailActivityInjectionTest`.

File: app/src/androidTest/java/com/skydoves/pokedex/DetailActivityInjectionTest.kt [hunk 2]
```diff
@@ -49,9 +50,13 @@ class DetailActivityInjectionTest {
...
-        activity.viewModel.pokemonInfoLiveData.observe(activity) { pokemonList ->
-          assertThat(pokemonList).isNotNull()
-        }
+
+        activity.viewModel.pokemonInfoLiveData.observe(
+          activity,
+          Observer { pokemonList ->
+            assertThat(pokemonList).isNotNull()
+          }
+        )
```
Comment (skydoves) [inline app/src/androidTest/java/com/skydoves/pokedex/DetailActivityInjectionTest.kt:59]: Is there any reason to use Observer? 
In kotlin 1.4.0, we can use SAM conversion and we don't need to declare a single abstract method in the trailing lambda.

Comment (yehezkiell) [reply app/src/androidTest/java/com/skydoves/pokedex/DetailActivityInjectionTest.kt:59]: oops, it was error in my local before. Then I update kotlin plugin to the latest and the error gone, I'll revert back

**Review by skydoves — APPROVED** [review]
LGTM :)

**Comment by sethepeterson** [comment]
Thanks for all the hard work :)
I'm having trouble integrating these changes into my project, specifically with AssistedInjectModule.kt::30.

`@Module(includes = [AssistedInject_AssistedInjectModule::class])`

produces this error:

> error: cannot find symbol
> @dagger.Module(includes = {AssistedInject_AssistedInjectModule.class})

I added the compile options to the app Gradle build, but am recieving that error. Did you guys have any issues with that? If so, how did you resolve the error?
Thank you!

**Pull request merged**